src/tools/rbd/action/Group.cc

```diff
@@ -756,6 +754,105 @@ int execute_group_snap_list(const po::variables_map &vm,
   return 0;
 }
 
+int execute_group_snap_info(const po::variables_map &vm,
+                            const std::vector<std::string> &ceph_global_args) {
+  size_t arg_index = 0;
+  std::string pool_name;
+  std::string namespace_name;
+  std::string group_name;
+  std::string group_snap_name;
+
+  int r = utils::get_pool_generic_snapshot_names(
+    vm, at::ARGUMENT_MODIFIER_NONE, &arg_index, at::POOL_NAME, &pool_name,
+    &namespace_name, GROUP_NAME, "group", &group_name, &group_snap_name, true,
+    utils::SNAPSHOT_PRESENCE_REQUIRED, utils::SPEC_VALIDATION_FULL);
+  if (r < 0) {
+    return r;
+  }
+
+  at::Format::Formatter formatter;
+  r = utils::get_formatter(vm, &formatter);
+  if (r < 0) {
+    return r;
+  }
+  Formatter *f = formatter.get();
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  r = utils::init(pool_name, namespace_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  librbd::group_snap_info2_t group_snap;
+  r = rbd.group_snap_get_info(io_ctx, group_name.c_str(),
+                              group_snap_name.c_str(), &group_snap);
+  if (r < 0) {
+    std::cerr << "rbd: failed to show group snapshot: "
+              << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  auto state_string = get_group_snap_state_name(group_snap.state);
+  if (f) {
+    f->open_object_section("group_snapshot");
+    f->dump_string("id", group_snap.id);
+    f->dump_string("name", group_snap.name);
+    f->dump_string("state", state_string);
+    f->dump_string("image_snap_name", group_snap.image_snap_name);
+    f->open_array_section("images");
+  } else {
+    std::cout << "rbd group snapshot '" << group_snap.name << "':\n"
+              << "\tid: " << group_snap.id << std::endl
+              << "\tstate: " << state_string << std::endl
+              << "\timage snap: " << group_snap.image_snap_name << std::endl
+              << "\timages:" << std::endl;
+  }
+
+  std::sort(group_snap.image_snaps.begin(), group_snap.image_snaps.end(),
+    [](const librbd::group_image_snap_info_t &lhs,
+       const librbd::group_image_snap_info_t &rhs) {
+      if (lhs.pool_id != rhs.pool_id) {
+        return lhs.pool_id < rhs.pool_id;
+      }
+      return lhs.image_name < rhs.image_name;
+    }
+  );
+
+  for (const auto& image_snap : group_snap.image_snaps) {
+    std::string pool_name;
+    r = rados.pool_reverse_lookup(image_snap.pool_id, &pool_name);
+    if (r < 0) {
+      pool_name = "<missing image pool " + stringify(image_snap.pool_id) + ">";
+    }
```

Comment:
Theoretically, other (transient) errors can be returned as well:
```suggestion
    if (r == -ENOENT) {
      pool_name = "<missing image pool " + stringify(image_snap.pool_id) + ">";
    } else if (r < 0) {
      std::cerr << "rbd: error looking up pool name for pool_id=" << pool_id
                << ": " << cpp_strerror(r) << std::endl;
      return r;
    }
```